### PR TITLE
[Feature] Add boolean variable type

### PR DESCRIPTION
**Pull request by Dvergar**
This PR adds the new boolean variable type.

Can be tested against https://studio.chiligrafx-dev.com/bench.html?workspace=latest&engine=293

## Related tickets
EDT-983
WRS-1201
WRS-1202

### CODE REVIEW

File: src/types/VariableTypes.ts [hunk 1]
```diff
@@ -33,21 +33,23 @@ export interface Variable {
     isVisible: boolean;
     isReadonly: boolean;
     isRequired: boolean;
+    occurrences: number;
 }
 
 export interface ImageVariable extends Variable {
-    occurrences: number;
     src?: ImageVariableSource;
```
Comment (olegsolodukhin) [inline src/types/VariableTypes.ts:40]: @NicolaVerbeeck Just noticed, that variable `value` field is called `src` for ImageVariable and `selected` for list variable. I think it is better to follow the same structure and call all of them `value`, isn't it?

File: types/Actions.d.ts [hunk 2]
```diff
@@ -167,6 +168,11 @@ declare module 'grafx-studio-actions' {
         selected?: string;
     }
 
+    export interface BooleanVariable extends BaseVariable {
+        type: VariableType.boolean;
+        value: boolean;
+    }
+
     interface GroupVariable extends BaseVariable {
         type: VariableType.group;
```
Comment (olegsolodukhin) [inline types/Actions.d.ts:177]: Sorry, just noticed, but why there is no `value` for this variable type? If it is present for all other types.

**Review by olegsolodukhin — APPROVED** [review]

File: types/Actions.d.ts (hunk 2 above)
Comment (Dvergar) [reply types/Actions.d.ts:177]: Indeed, group is just a holder variable to group children. It has no value, children refer to the group by its id.

File: src/types/VariableTypes.ts (hunk 1 above)
Comment (Dvergar) [reply src/types/VariableTypes.ts:40]: I agree it's not consistent. And I don't know if we want to change this now. It means migration > breaking change > WRS has to update 🥲

 For the list, though, what would be the value: items or selected?

Comment (Dvergar) [reply src/types/VariableTypes.ts:40]: @Matthiee @NicolaVerbeeck @evan-mcgeek @olegsolodukhin 

Not in scope of this ticket but can I have your input on this:

Maybe indeed `ImageVariable.src` should be `ImageVariable.value` & `ListVariable.items/selected` should be `ImageVariable.value` to be consistent with the other variables.

If we agree, two thoughts comes to my mind:
- It's enough w/ the breaking-changes for now
- If we are making breaking-changes, it's better to have all of them condensed in the same time frame

WDYT?

Comment (NicolaVerbeeck) [reply src/types/VariableTypes.ts:40]: ListVariable.value would then be the new name for ListVariable.selected?

**Review by NicolaVerbeeck — APPROVED** [review]

**Review by evan-mcgeek — APPROVED** [review]

File: src/types/VariableTypes.ts (hunk 1 above)
Comment (olegsolodukhin) [reply src/types/VariableTypes.ts:40]: for the list - value is selected. Similarity: boolean has two options - true/false, the value is the selected one; String - all possible combinations of letters (possible), the value is the "selected" one; for the list we define possible values and value refers to the chosen one.

**Review by srares76 — APPROVED** [review]

**Pull request merged**